asQ/allatoncesystem.py

```diff
@@ -479,7 +480,6 @@ def get_test(i):
             aao_form -= (1.0/dt)*self.form_mass(*w0s, *dws)
 
             # vector field
-            aao_form += theta*self.form_function(*w1s, *dws)
-            aao_form += (1-theta)*self.form_function(*w0s, *dws)
-
+            aao_form += theta*self.form_function(*w1s, *dws, self.time[n])
+            aao_form += (1-theta)*self.form_function(*w0s, *dws, self.time[n])
```

Comment:
The call to `form_function` on line 484 needs to give `self.time[n-1]` because it is the function evaluated at the _previous_ timestep.
For the first timestep in the window this needs to be the time at the window's initial condition, so I think we need another `Constant` for this initial time that we can update at each window.